### ISSUE TITLE
Collation error on BOOLEAN and NUMBER(38,0) type columns

### ISSUE THREAD

**Issue by MyJBMe** [Snowflake-Labs/terraform-provider-snowflake#2657]
### Terraform CLI and Provider Versions

0.87.0, 0.87.1, 0.87.2

### Terraform Configuration

```terraform
terraform {
  required_providers {
    snowflake = {
      source  = "Snowflake-Labs/snowflake"
      version = "~> 0.86.0"
    }
  }
backend "s3" {}

}
```

### Expected Behavior

Deployment of all tables without any issues.
As far as I understand column collation, it should not apply to non-text columns, doesn't it? At least it seems to be about text (documentation states: "Collation allows you to specify alternative rules for comparing strings", https://docs.snowflake.com/en/sql-reference/collation#label-collation-definition). Thereby I wonder about getting issues for different datatypes, since I would expect the provider to not apply collation to non-text columns. 

### Actual Behavior

Error: error changing property on DATABASE|SCHEMA|TABLE: err 001478 (42601): SQL compilation error:
│ Cannot specify column collation for data type 'BOOLEAN' for column 'COLUMN9'
│ 
│   with module.xxx.snowflake_table.table_name,
│   on stack/module_name/tables_base.tf line 57, in resource "snowflake_table" "table_name"

Error: error changing property on DATABASE|SCHEMA|TABLE: err 001478 (42601): SQL compilation error:
│ Cannot specify column collation for data type 'NUMBER(38,0)' for column 'ID'
│ 
│   with module.xxx.snowflake_table.table_name,
│   on stack/module_name/tables_base.tf line 67, in resource "snowflake_table" "table_name"

### Steps to Reproduce

1. Create .tf file with resource:

resource "snowflake_table" "table_name" {
  database            = var.DATABASE_NAME
  schema              = var.SCHEMA_NAME
  name                = var.TABLE_NAME
  comment             = "A table containing [...]"

  change_tracking = true

  column {
    name = "ID"
    type = "NUMBER(38,0)"
  }

  column {
    name = "AVAILABLE"
    type = "BOOLEAN"
  }
}

2. `terraform apply`
3. Issues should occur after the second deployment (e.g. after updating existing resources)

### How much impact is this issue causing?

Medium

**Comment by sfc-gh-asawicki** [comment]
Hey @MyJBMe. Thanks for reaching out to us.

It looks like a duplicate of #2588. It was fixed in #2607. It was not officially released yet, but it is available in the release candidate v0.87.3-pre. It's available in the registry but you have to specify the whole version string to use it.

**Comment by sfc-gh-asawicki** [comment]
Closing as it was fixed in #2607. Please open a new issue if the problem persists in the newest provider version.
